tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,888 @@
+        entry_point=os.path.join(_XGBOOST_PATH, "inference.py"),
+        enable_network_isolation=True,
+    )
+    step_args = xgb_model.register(
+        content_types=["application/json"],
+        response_types=["application/json"],
+        inference_instances=["ml.m5.large"],
+        transform_instances=["ml.m5.large"],
+        model_package_group_name=model_package_group_name,
+        approval_status="Approved",
+    )
+    step_model = ModelStep(
+        name="MyModelStep",
+        step_args=step_args,
+    )
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[xgb_model_data_param],
+        steps=[step_model],
+        sagemaker_session=pipeline_session,
+    )
+    try:
+        pipeline.create(role)
+
+        for _ in retries(
+            max_retry_count=5,
+            exception_message_prefix="Waiting for a successful execution of pipeline",
+            seconds_to_sleep=10,
+        ):
+            execution = pipeline.start(parameters={})
+            try:
+                execution.wait(delay=30, max_attempts=60)
+            except WaiterError:
+                pass
+
+            # Verify the pipeline execution succeeded
+            step_register_model = None
+            execution_steps = execution.list_steps()
+            is_execution_fail = False
+            for step in execution_steps:
+                failure_reason = step.get("FailureReason", "")
+                if failure_reason != "":
+                    logging.error(
+                        f"Pipeline execution failed with error: {failure_reason}." " Retrying.."
+                    )
+                    is_execution_fail = True
+                    break
+                assert step["StepStatus"] == "Succeeded"
+                if _REGISTER_MODEL_NAME_BASE in step["StepName"]:
+                    step_register_model = step
+            if is_execution_fail:
+                continue
+            assert len(execution_steps) == 2
+            assert step_register_model
+            break
+
+        # Verify the registered model can work as expected
+        with timeout_and_delete_endpoint_by_name(
+            endpoint_name=endpoint_name, sagemaker_session=pipeline_session
+        ):
+            model_pkg_arn = step_register_model["Metadata"][_REGISTER_MODEL_TYPE]["Arn"]
+            response = pipeline_session.sagemaker_client.describe_model_package(
+                ModelPackageName=model_pkg_arn
+            )
+            model_package = ModelPackage(
+                role=role,
+                model_data=response["InferenceSpecification"]["Containers"][0]["ModelDataUrl"],
+                model_package_arn=model_pkg_arn,
+                sagemaker_session=sagemaker_session,
+            )
+            model_package.deploy(
+                initial_instance_count=1,
+                instance_type="ml.m5.large",
+                endpoint_name=endpoint_name,
+            )
+            predictor = XGBoostPredictor(
```

Comment:
Note that this test can still pass even if the script mode env vars weren't added to model registry (it might just be using the normal container inference).

Can we describe the model package and ensure that `SAGEMAKER_SUBMIT_DIRECTORY` and `SAGEMAKER_PROGRAM` were set correctly? I would say you could describe the model for that, but I'm pretty sure it will just return the ModelPackageArn where the container definition would be.

Reply:
Let me know if this is validated somewhere else and I missed it

Reply:
> Note that this test can still pass even if the script mode env vars weren't added to model registry

No. If the script is not run, the predict response[0] data length would be different (i.e. 1). As you can see below the length check is actually validating the script run:
```
# extra data are appended by the inference.py
            assert len(response[0]) > 1
 ```
 
Previously I had 3 E2E tests for the XGBoost model and one case is no entry_point or script ENVs at all. In that case, the len(response[0]) == 1.
However, considering the limited endpoint resources, I removed the extra E2E tests.

Reply:
Plus all the model step unit tests are validating the container env vars fetched from the Pipeline definition, so it should be good.